APITesters/SwiftAPITester/SwiftAPITester/PurchasesAPI.swift

```diff
@@ -0,0 +1,198 @@
+    let isPurchasesConfigured: Bool = Purchases.isConfigured
+
+    print(canI, version, automaticAppleSearchAdsAttributionCollection, debugLogsEnabled, logLevel, proxyUrl!,
+          forceUniversalAppStore, simulatesAskToBuyInSandbox, sharedPurchases, isPurchasesConfigured)
+}
+
+private func checkPurchasesPurchasingAPI(purchases: Purchases) {
+    let piComplete: ReceivePurchaserInfoBlock = { _, _ in }
+    purchases.purchaserInfo(piComplete)
+    purchases.purchaserInfo { _, _ in }
+
+    let offeringsComplete: ReceiveOfferingsBlock = { _, _ in }
+    purchases.offerings(offeringsComplete)
+    purchases.offerings { _, _ in }
+
+    let productsComplete: ReceiveProductsBlock = { _ in }
+    purchases.products([String](), productsComplete)
+    purchases.products([String]()) { _ in }
+
+    let skp: SKProduct = SKProduct()
+    let skpd: SKProductDiscount = SKProductDiscount()
+    let skmd: SKPaymentDiscount = SKPaymentDiscount()
+    let pack: Package! = nil
+
+    let purchaseProductComplete: PurchaseCompletedBlock = { _, _, _, _  in }
+    purchases.purchase(product: skp, purchaseProductComplete)
+    purchases.purchase(product: skp) { _, _, _, _  in }
+    purchases.purchase(package: pack, purchaseProductComplete)
+    purchases.purchase(package: pack) { _, _, _, _  in }
+
+    purchases.restoreTransactions(piComplete)
+    purchases.syncPurchases(piComplete)
+
+    let checkEligComplete: ([String: IntroEligibility]) -> Void = { _ in }
+    purchases.checkTrialOrIntroductoryPriceEligibility([String](), completion: checkEligComplete)
+    purchases.checkTrialOrIntroductoryPriceEligibility([String]()) { _ in }
+
+    let discountComplete: PaymentDiscountBlock = { _, _ in }
+
+    purchases.paymentDiscount(forProductDiscount: skpd, product: skp, completion: discountComplete)
+    purchases.paymentDiscount(forProductDiscount: skpd, product: skp) { _, _ in }
+
+    purchases.purchase(product: skp, discount: skmd, completion: purchaseProductComplete)
+    purchases.purchase(product: skp, discount: skmd) { _, _, _, _  in }
+    purchases.purchase(package: pack, discount: skmd, completion: purchaseProductComplete)
+    purchases.purchase(package: pack, discount: skmd) { _, _, _, _  in }
+    purchases.invalidatePurchaserInfoCache()
+
+    // PurchasesDelegate
+    let purchaserInfo: PurchaserInfo? = nil
+    purchases.delegate?.purchases?(purchases, didReceiveUpdated: purchaserInfo!)
+
+    let defermentBlock: DeferredPromotionalPurchaseBlock = { _ in }
+    purchases.delegate?.purchases?(purchases, shouldPurchasePromoProduct: skp, defermentBlock: defermentBlock)
+    purchases.delegate?.purchases?(purchases, shouldPurchasePromoProduct: skp) { _ in }
+}
+
+private func checkIdentity(purchases: Purchases) {
+    let piComplete: ReceivePurchaserInfoBlock = { _, _ in }
+
+    // should have deprecation warning 'createAlias' is deprecated: Use logIn instead.
+    purchases.createAlias("", piComplete)
+    purchases.createAlias("") { _, _ in }
```

Comment:
I see, this is the kind of thing why we wanted to separate the api testers into a separate project, right? 
otherwise, folks using Xcode direct integration would have to deal with these

Reply:
Yeah, I believe they might end up needing to deal with the warnings.

Reply:
they wouldn't, though, if we move our API testers into a separate project, right? 
we can have them live in a separate project and then create an xcworkspace to incorporate them all, so that we can still easily test stuff, while making it easy for other folks to integrate

Reply:
So, one would think this is right, but it appears that if we use SPM in one project for a dependency in the workspace, and not in another, then Xcode doesn't like that. Right now the APITesters use SPM for their dependency. I don't remember the motivation behind that, but if we were to just rely on the RevenueCat.xcodeproj in a workspace instead, then 100% you're right.

Reply:
https://github.com/RevenueCat/purchases-ios/issues/857

Reply:
As of now they just use SPM because it's the first integration I tried 🤷  I was considering trying to create testers using carthage and cocoapods as well, at least for testing the migration guide locally. That's where @taquitos was helping me figure out how to have a shared folder between different xcode projects, so hopefully only the integration piece needs to be separate.

Longer term, I see the API testers as purely testing our API...there wouldn't be any differences in our API between integrations, right?